benchmarks/src/main/kotlin/macrobenchmarks/ChannelProducerConsumerMonteCarloBenchmark.kt

```diff
@@ -0,0 +1,264 @@
+ * the current iteration.
+ *
+ * The benchmark stops execution the current benchmark configuration if execution times became stable and stopped changing
+ * drastically.
+ */
+fun main() {
+    // Create a new output CSV file and write the header
+    Files.createDirectories(Paths.get(OUTPUT).parent)
+    writeOutputHeader()
+
+    val totalIterations = ChannelCreator.values().size * THREADS.size * WITH_BALANCING.size *
+                                                DispatcherCreator.values().size * WITH_SELECT.size
+    var currentConfigurationNumber = 0
+    val startTime = System.currentTimeMillis()
+
+    for (channel in ChannelCreator.values()) {
+        for (threads in THREADS) {
+            for (withBalancing in WITH_BALANCING) {
+                for (dispatcherType in DispatcherCreator.values()) {
+                    for (withSelect in WITH_SELECT) {
+                        val args = arrayOf(channel.toString(),
+                                threads.toString(),
+                                withBalancing.toString(),
+                                dispatcherType.toString(),
+                                withSelect.toString(),
+                                currentConfigurationNumber.toString(),
+                                totalIterations.toString(),
+                                startTime.toString())
+                        val exitValue = runProcess(MonteCarloIterationProcess::class.java.name, jvmOptions, args)
+                        if (exitValue != 0) {
+                            println("The benchmark couldn't complete properly, will end running benchmarks")
+                            return
+                        }
+                        currentConfigurationNumber++
+                    }
+                }
+            }
+        }
+    }
+}
+
+class MonteCarloIterationProcess {
+    companion object {
+        @JvmStatic
+        fun main(args: Array<String>) {
+            val channel = ChannelCreator.valueOf(args[0])
+            val threads = args[1].toInt()
+            val withBalancing = args[2].toBoolean()
+            val dispatcherType = DispatcherCreator.valueOf(args[3])
+            val withSelect = args[4].toBoolean()
+            val currentConfigurationNumber = args[5].toInt()
+            val totalIterations = args[6].toInt()
+            val startTime = args[7].toLong()
+
+            print("\rchannel=$channel threads=$threads withBalancing=$withBalancing dispatcherType=$dispatcherType withSelect=$withSelect: warm-up phase... [${eta(currentConfigurationNumber, totalIterations, startTime)}]")
+
+            repeat(WARM_UP_ITERATIONS) {
+                runIteration(threads, channel, withBalancing, withSelect, dispatcherType)
+            }
+
+            runMonteCarlo(threads, channel, withBalancing, withSelect, dispatcherType) {
+                eta(currentConfigurationNumber, totalIterations, startTime)
```

Comment:
Please also print benchmark score for each intermediate configuration (in addition to configuration and ETA itself)

Reply:
we already print this information